src/index.js

```diff
@@ -1,21 +1,17 @@
 "use strict";
 
-let WritableStream = require("./writable_stream");
+const WritableStream = require("./writable_stream");
 
-module.exports = class Renderer {
-	constructor(bundlePath) {
-		this.bundlePath = bundlePath;
-		this.reload();
-	}
+module.exports = bundle => {
+	const _render = bundle;
 
-	render(response, tag, params) {
-		// TODO: `response.status(200);`?
-		let stream = new WritableStream(response);
-		this._render(stream, tag, params);
-		response.end();
-	}
+	return (req, res, next) => {
+		res.render = (tag, params) => {
```

Comment:
tried to go with the standard mechanism of `app.set('view engine')` to make it work as designed, but then the first param has to be a filePath, that actually has to exist. So adding it via middleware-override is already a hack, so yes you may be right. 

`res.complate('some-component')` should make it more clear?!